### ISSUE TITLE
test fails when run in parallel

### ISSUE THREAD

**Issue by ykaliuta** [opendatahub-io/opendatahub-operator#739]
**Describe the bug**

I observed several times `make unit-test` failing in github with message `DataScienceCluster initialization Handling existing resources [AfterEach] Should not have more than one DSCI instance in the cluster` (https://github.com/opendatahub-io/opendatahub-operator/actions/runs/6868213399/job/18678252151?pr=736) while it runs fine on the local machine. When I tried parallel run locally (directly `ginkgo -p 2 ./controllers/... ./tests/integration/features/...` because `go test` complains:
```
Ginkgo detected configuration issues:
Use of go test -parallel
  Go test's implementation of parallelization does not actually parallelize
  Ginkgo specs.  Please use the ginkgo cli and `ginkgo -p` or `ginkgo -procs=N`
  instead.
```
) it fails as well like
```
• [FAILED] [1.070 seconds]
DataScienceCluster initialization Handling existing resources [It] Should not update configmap if it exists
/home/ykaliuta/work/rhods/opendatahub-operator/controllers/dscinitialization/dscinitialization_test.go:171
```

**To Reproduce**
KUBEBUILDER_ASSETS=$PWD/bin/k8s/1.24.2-linux-amd64 ginkgo -p 2 ./controllers/... ./tests/integration/features/...

**Expected behavior**
Test passes

**Comment by ykaliuta** [comment]
After some investigation I think it's probably unrelated.

**Comment by bartoszmajsak** [comment]
Tests for DSCI are flaky, they've been failing for me in random places and environments. Right now I'm frequently seeing following:

```
  [FAILED] in [AfterEach] - /home/bartek/code/rhods/opendatahub-operator/controllers/dscinitialization/dscini
tialization_test.go:244 @ 11/21/23 08:50:07.867                                                              
  << Timeline                                                                                                
                                                                                                             
  [FAILED] Timed out after 20.001s.                                                                          
  Expected                                                                                                   
      <bool>: false                                                                                          
  to be true                                                                                                 
  In [AfterEach] at: /home/bartek/code/rhods/opendatahub-operator/controllers/dscinitialization/dscinitializa
tion_test.go:244 @ 11/21/23 08:50:07.867                                                                     
------------------------------                                                                               
•••••••••                                                                                                    
                                                                                                             
Summarizing 1 Failure:                                                                                       
  [FAIL] DataScienceCluster initialization Creation of related resources [AfterEach] Should create default ap
plication namespace                                                                                          
  /home/bartek/code/rhods/opendatahub-operator/controllers/dscinitialization/dscinitialization_test.go:244   
                                                                                                             
Ran 10 of 10 Specs in 31.910 seconds                                                                         
FAIL! -- 9 Passed | 1 Failed | 0 Pending | 0 Skipped     
```


I think we should rework this test file and make sure we do not have any side effects and leftovers affecting execution of other tests.

@zdtsw should we open new issue for flakiness or rename this one?

@ykaliuta As for parallelization - this will need some extra work, as I doubt current DSCI tests are not relying on a sequential execution.

**Comment by AjayJagan** [comment]
As for flakiness, I guess we should start using different namespaces? and this might also help with running tests in parallel.

**Comment by ykaliuta** [comment]
> @ykaliuta As for parallelization - this will need some extra work, as I doubt current DSCI tests are not relying on a sequential execution.

Yes, it's different issue. Let's create another one for flakiness.

**Comment by AjayJagan** [comment]
JIRA issue: https://issues.redhat.com/browse/RHOAIENG-411

**Comment by AjayJagan** [comment]
migrated to JIRA, hence closing this issue.
